### PR TITLE
Fix deadlock inside BASS library finalization.

### PR DESCRIPTION
**Pull request by MiranDMC**

### CODE REVIEW

File: cleo_plugins/Audio/CSoundSystem.cpp [hunk 1]
```diff
@@ -52,7 +53,7 @@ namespace CLEO
         if (initialized)
         {
             TRACE("Freeing BASS library");
-            BASS_Free();
+            std::thread(BASS_Free);
```
Comment (x87) [inline cleo_plugins/Audio/CSoundSystem.cpp:56]: why is there a deadlock?

Comment (MiranDMC) [reply cleo_plugins/Audio/CSoundSystem.cpp:56]: The library internally waits to some (COM?) objects to be finalized. BASS author says free can not be called in dll main. I tried to do it from some other callback like on game closing, but with same results. Other author suggestion is not to call free at all and let it be released automatically, but that makes game crashing (I guess due all ModLoader hacks).

Calling it from another thread works. It is visible in log that CLEO proceeded with other cleanup steps, game aslo closes without problems.

Comment (x87) [reply cleo_plugins/Audio/CSoundSystem.cpp:56]: Interesting. I would also remove free and let OS handle the unloading.

Is the crash reproducible 100%? Does it depend on other plugins installed in game?

Comment (MiranDMC) [reply cleo_plugins/Audio/CSoundSystem.cpp:56]: Seems like all you need is ModLoader

Comment (x87) [reply cleo_plugins/Audio/CSoundSystem.cpp:56]: Why it wasn't an issue in CLEO4?

Comment (MiranDMC) [reply cleo_plugins/Audio/CSoundSystem.cpp:56]: I have no idea. ModLoader is thin ice. In CLEO4 Bass was managed by core.

Comment (MiranDMC) [reply cleo_plugins/Audio/CSoundSystem.cpp]: Hm seems to be working without calling free at all. I have no idea why game was crashing for me when testing it previously.

**Review by x87 — APPROVED** [review]

**Pull request merged**